File: 2023/D08/part1.hs
#file = open("test.txt", "r")
file = open("input.txxt", "r")

lines = [l.strip() for l in file if l != "\n"]
moves = lines[0]

dicc = dict()
lines = lines[1:]
for line in lines:
    node, neighbors = line.split(" = ")
    n1, n2 = neighbors[1:len(neighbors) - 1].split(", ")
    dicc[node] = [n1, n2]

node = "AAA"
ans = 0
move = 0
while 1:
    if node == "ZZZ":
        break
    node = dicc[node][0] if moves[move] == 'L' else dicc[node][1]
    move = (move + 1) % len(moves)
    ans += 1
print(ans)

File: 2022/D04/part1.js
lines = []
require("fs")
  .readFileSync("input.txt").toString()
  .split(/\r?\n/)
  .forEach( (line) => {
    lines.push(line)
  }
);

let count = 0
lines.map((line) => {
	line = line.split(",")
	L1 = parseInt(line[0].split("-")[0])
	R1 = parseInt(line[0].split("-")[1])
	L2 = parseInt(line[1].split("-")[0])
	R2 = parseInt(line[1].split("-")[1])
	
	if(L1 <= L2 && R2 <= R1) count++
	else if(L1 >= L2 && R2 >= R1) count++
})

console.log(count)